service/authorization/authorization.go

```diff
@@ -282,12 +287,23 @@ func (as *AuthorizationService) GetDecisions(ctx context.Context, req *authoriza
 
 					// TODO this might cause errors if multiple entities dont have ids
 					// currently just adding each entity returned to same list
-					for _, e := range ecEntitlements.GetEntitlements() {
+					for idx, e := range ecEntitlements.GetEntitlements() {
+						entityID := e.GetEntityId()
+						if entityID == "" {
+							entityID = EntityIDPrefix + fmt.Sprint(idx)
+						}
 						auditECEntitlements = append(auditECEntitlements, audit.EntityChainEntitlement{
-							EntityID:                 e.GetEntityId(),
+							EntityID:                 entityID,
 							AttributeValueReferences: e.GetAttributeValueFqns(),
 						})
-						entityAttrValues[e.GetEntityId()] = e.GetAttributeValueFqns()
+						entityCategory := entities[idx].GetCategory()
+
+						// If entity type unspecified, include in access decision to err on the side of caution
```

Comment:
lmk if this is the right behavior

Reply:
I think its better to be safer here and check unspecified.